File: R/8kyu/8kyu-array-plus-array.r
# Array plus array
# Language: R
# Difficulty: 8 kyu

array_plus_array <- function(vec1,vec2){
  sum(c(vec1, vec2))
}


# Tests
test_that('Basic tests', {
  expect_equal(array_plus_array(c(1, 2, 3), c(4, 5, 6)), 21)
  expect_equal(array_plus_array(c(-1, -2, -3), c(-4, -5, -6)), -21)
  expect_equal(array_plus_array(c(0, 0, 0), c(4, 5, 6)), 15)
  expect_equal(array_plus_array(c(100, 200, 300), c(400, 500, 600)), 2100)
})

File: Julia/8kyu/8kyu-array-plus-array.jl
# Array plus array
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

function arrayplusarray(arr1, arr2)
  sum(vcat(arr1, arr2))
end


# Tests
using FactCheck

facts("Basic tests") do
  context("Should pass fixed tests") do
    @fact arrayplusarray([1, 2, 3], [4, 5, 6]) --> 21
    @fact arrayplusarray([-1, -2, -3], [-4, -5, -6]) --> -21
    @fact arrayplusarray([0, 0, 0], [4, 5, 6]) --> 15
    @fact arrayplusarray([100, 200, 300], [400, 500, 600]) --> 2100
  end
end